crates/compiler/can/src/constraint.rs

```diff
@@ -773,6 +785,8 @@ pub enum Constraint {
     /// Attempt to resolve a specialization.
     Resolve(OpportunisticResolve),
     CheckCycle(Index<Cycle>, IllegalCycleMark),
+
+    IngestedFile(TypeOrVar, Box<Path>, Arc<Vec<u8>>),
```

Comment:
I suspect `Arc<Vec<u8>>` can be `Arc<[u8]>`, saving an allocation and level of indirection.

Reply:
Doing so will make `Constraint` even bigger. `Arc<[u8]>` is 16 bytes. `Arc<Vec<u8>>` is 8 bytes.